src/modules/0.0.21/aws_dynamo/index.ts

```diff
@@ -102,27 +103,32 @@ class DynamoTableMapper extends MapperBase<DynamoTable> {
       return out;
     },
     read: async (ctx: Context, id?: string) => {
-      const client = (await ctx.getAwsClient()) as AWS;
+      const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
       if (id) {
-        const rawTable = await this.getTable(client.dynamoClient, id);
-        if (!rawTable?.Table) return;
-        return this.dynamoMapper(rawTable.Table);
+        for (const region of enabledRegions) {
+          const client = (await ctx.getAwsClient(region)) as AWS;
+          const rawTable = await this.getTable(client.dynamoClient, id);
+          if (!rawTable?.Table) continue;
+          return this.dynamoMapper(rawTable.Table, region);
```

Comment:
shouldn't we be collecting these and then returning all of them outside the for-loop?

Reply:
This is on the "request by ID" path, where the ID is the dynamoDB table name, but we don't have any information on what region that particular table is located in, so we check them in order.

But the only way for this query to make sense is if there is only one table with that name across all regions.

That assumption may not hold up, but I am not super worried at the moment because I don't believe anything uses the "cloud query dynamo table by name" path at all, so it's only included for fulfilling the typescript types.

Reply:
got it